dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ranger/RangerClient.java

```diff
@@ -0,0 +1,208 @@
+      return new AutoValue_RangerClient_ListGroupsResponse(vXGroups);
+    }
+
+    @JsonProperty
+    abstract ImmutableList<Group> vXGroups();
+  }
+
+  @AutoValue
+  abstract static class ListRolesResponse {
+
+    @JsonCreator
+    static ListRolesResponse create(@JsonProperty("roles") ImmutableList<Role> roles) {
+      return new AutoValue_RangerClient_ListRolesResponse(roles);
+    }
+
+    @JsonProperty
+    abstract ImmutableList<Role> roles();
+  }
+
+  @AutoValue
+  abstract static class ListServicesResponse {
+
+    @JsonCreator
+    static ListServicesResponse create(@JsonProperty("services") ImmutableList<Service> services) {
+      return new AutoValue_RangerClient_ListServicesResponse(services);
+    }
+
+    @JsonProperty
+    abstract ImmutableList<Service> services();
+  }
+
+  @AutoValue
+  abstract static class ListPoliciesResponse {
+
+    @JsonCreator
+    static ListPoliciesResponse create(@JsonProperty("policies") ImmutableList<Policy> policies) {
+      return new AutoValue_RangerClient_ListPoliciesResponse(policies);
+    }
+
+    @JsonProperty
+    abstract ImmutableList<Policy> policies();
+  }
+
+  public static class RangerException extends Exception {
+
+    public RangerException(String message, Throwable cause) {
+      super(message, cause);
+    }
+
+    public RangerException(String message) {
+      super(message);
+    }
+  }
+
+  private final URI baseUri;
+
+  private final CloseableHttpClient httpClient;
+
+  private final UsernamePasswordCredentials credentials;
+
+  public RangerClient(URI baseUri, String user, String password) throws URISyntaxException {
+    this.baseUri = baseUri;
+    httpClient = HttpClients.createMinimal();
+    credentials = new UsernamePasswordCredentials(user, password);
+  }
+
+  public ImmutableList<User> findUsers(Map<String, String> args) throws RangerException {
+    return doHttpGet("/service/xusers/users", args, ListUsersResponse.class).vXUsers();
+  }
+
+  public ImmutableList<Group> findGroups(Map<String, String> args) throws RangerException {
+    return doHttpGet("/service/xusers/groups", args, ListGroupsResponse.class).vXGroups();
+  }
+
+  public ImmutableList<Role> findRoles(Map<String, String> args) throws RangerException {
+    return doHttpGet("/service/roles/roles", args, ListRolesResponse.class).roles();
+  }
+
+  public ImmutableList<Service> findServices(Map<String, String> args) throws RangerException {
+    return doHttpGet("/service/plugins/services", args, ListServicesResponse.class).services();
+  }
+
+  public ImmutableList<Policy> findPolicies(Map<String, String> args) throws RangerException {
+    return doHttpGet("/service/plugins/policies", args, ListPoliciesResponse.class).policies();
+  }
+
+  private <T> T doHttpGet(String path, Map<String, String> params, Class<T> bodyClass)
+      throws RangerException {
+    URI uri;
+    try {
+      uri =
+          new URIBuilder(baseUri + path)
```

Comment:
I would suggest to either do:
```
new URIBuilder(baseUri).setPath(path)
```

or if the baseUri can also contain path, then add a static method that would concatenate `baseUri` and `path` making sure that a single slash is added in between no matter if they start/end with a slash or not.

Reply:
Done.